syntax = "proto3";

option java_multiple_files = true;

package org.order.service;

import "google/protobuf/timestamp.proto";

message Order {
  int32 order_id = 1;
  int32 user_id = 2;
  google.protobuf.Timestamp order_date = 3;
  int32 total_amount = 4;
}

message OrdersByUserIdRequest {
  int32 user_id = 1;
}

message OrdersByUserIdResponse {
  repeated Order orders = 1;
}

service OrderService {
  rpc GetOrdersByUserId(OrdersByUserIdRequest) returns (OrdersByUserIdResponse);
}